packages/contracts/foundry_tests/IncentivesController3.t.sol

```diff
@@ -0,0 +1,404 @@
+        vm.label(0x6324511D46c6339D49697C23ca35cde62089E32a, "DepositWithdrawLogic");
+        vm.label(address(ASSET_MAPPINGS), "ASSET_MAPPINGS");
+        vm.label(OP_TEST_USER, "OPTIMISM_TEST_USER");
+        vm.label(address(incentivesController), "INCENTIVES_CONTROLLER");
+    }
+
+    function _createUsers() internal {
+        users.push(makeAddr("alice"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(365 days);
+
+        users.push(makeAddr("bob"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(2 * 365 days);
+
+        users.push(makeAddr("eve"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(3 * 365 days);
+
+        users.push(makeAddr("joe"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(4 * 365 days);
+
+        users.push(makeAddr("mike"));
+        usdcDepositAmounts.push(1e9);
+        wethDepositAmounts.push(1 ether);
+        vmexLpLockAmounts.push(1e20);
+        vmexLpLockDurations.push(5 * 365 days);
+    }
+
+    function testUpgrade() public {
+        bytes32 oldImpl = vm.load(addressesProvider.getIncentivesController(), IMPLEMENTATION_SLOT);
+
+        _upgradeIncentivesController();
+
+        assertFalse(oldImpl == vm.load(addressesProvider.getIncentivesController(), IMPLEMENTATION_SLOT));
+        assertEq(incentivesController.VE_VMEX(), VE_VMEX);
+        assertEq(incentivesController.DVMEX_REWARD_POOL(), DVMEX_REWARD_POOL);
+        assertEq(incentivesController.DVMEX(), DVMEX);
+    }
+
+    function testQueueNewRewards() public {
+        _upgradeIncentivesController();
+
+        DataTypes.ReserveData memory reserveData = lendingPool.getReserveData(TOKEN_USDC, LP_TRANCHE_ID);
+        _queueNewRewards(reserveData.aTokenAddress, REWARD_AMOUNT);
+
+        IncentivesController.DVmexReward memory reward = incentivesController.getDVmexReward(reserveData.aTokenAddress);
+
+        assertEq(reward.periodFinish, uint32(block.timestamp + 14 days));
+        assertEq(reward.lastUpdateTime, uint32(block.timestamp));
+        assertEq(reward.rewardRate, uint64(REWARD_AMOUNT / 14 days));
+        assertEq(reward.rewardPerTokenStored, 0);
+        assertEq(reward.queuedRewards, 0);
+        assertEq(reward.currentRewards, uint128(REWARD_AMOUNT));
+    }
+
+    function testClaimMinBoost() public {
+        _upgradeIncentivesController();
+
+        DataTypes.ReserveData memory reserveData = lendingPool.getReserveData(TOKEN_USDC, LP_TRANCHE_ID);
+
+        _queueNewRewards(reserveData.aTokenAddress, REWARD_AMOUNT);
+
+        _deposit(USER, TOKEN_USDC, LP_TRANCHE_ID, 1e9);
```

Comment:
Since there is no vmexLp locked, then wouldn't this be actually testing max boost?

You probably need to modify this test to lock some vmexLp <= baseRewards to see what happens. Make sure to test both branches of this:
```
        return _min(
            (
                aTokenBalance * BOOSTING_FACTOR
                    + _standardizeDecimals(aTokenSupply, decimals) * IVeVmex(VE_VMEX).balanceOf(_account) / veTotalSupply
                        * (BOOST_DENOMINATOR - BOOSTING_FACTOR)
            ) / BOOST_DENOMINATOR,
            aTokenBalance
        );
```

Reply:
I meant when the veTotalSupply = 0

```
        uint256 veTotalSupply = IVeVmex(VE_VMEX).totalSupply();
        if (veTotalSupply == 0) {
            return aTokenBalance;
        }
```